earth_enterprise/src/fusion/autoingest/AssetFactory.h

```diff
@@ -256,6 +257,196 @@ namespace AssetFactory
+      {
+          asset = Make<AssetHandleType, ConfigType>
+                  (ref_, type_, meta_, config_);
+      }
+      bool needed = false;
+      return asset->MyUpdate(needed);
+  }
+
+  template <class MutableDerivedVersionHandleType, class Version,
+            class ConfigType, class VersionDType>
+  MutableDerivedVersionHandleType ReuseOrMakeAndUpdate(const std::string& ref_,
+                                                       AssetDefs::Type type_,
+                                                       const std::vector<SharedString>& inputs_,
+                                                       const khMetaData& meta_,
+                                                       const ConfigType& config_,
+                                                       const std::vector<Version>& cachedinputs_)
+  {
+      using AssetHandleType = typename MutableDerivedVersionHandleType::Impl::MutableAssetType;
+      std::vector<SharedString> inputarg { inputs_ }, boundInputs(inputarg.size());
+
+      std::transform(inputarg.begin(), inputarg.end(), back_inserter(boundInputs),
+                     ptr_fun(&AssetVersionRef::Bind));
+
+      auto asset = Find<AssetHandleType>(ref_, type_);
+      if (asset)
+      {
+          for (const auto& v : asset->versions)
+          {
+              try
+              {
+                  VersionDType version(v);
+                  version.LoadAsTemporary();
+
+                  if ((version->state != AssetDefs::Offline) &&
+                     (version->inputs == boundInputs) &&
+                     config_.IsUpToDate(version->config))
+                  {
+                      version.MakePermanent();
+                      return version;
+                  }
+                  else
+                  {
+                      version.NoLongerNeeded();
+                  }
+              }
+              catch (...)
+              {
+                  notify(NFY_WARN,"could not reuse a version");
+              }
+          }
+
+          asset->Modify(inputs_, meta_, config_);
+      }
+      else
+      {
+          asset = Make<AssetHandleType, ConfigType>
+                  (ref_, type_, inputs_, meta_, config_);
+      }
+      bool needed = false;
+      return asset->MyUpdate(needed, cachedinputs_);
+  }
+
+  template <class MutableDerivedVersionHandleType,
+            class ConfigType, class VersionDType, class Extras>
+  MutableDerivedVersionHandleType ReuseOrMakeAndUpdate(const std::string& ref_,
+                                                       AssetDefs::Type type_,
+                                                       const khMetaData& meta_,
+                                                       const ConfigType& config_,
+                                                       const Extras& extraArgs)
+  {
+      using AssetHandleType = typename MutableDerivedVersionHandleType::Impl::MutableAssetType;
+      std::vector<SharedString> inputarg, boundInputs;
```

Comment:
Not a big deal, but somehow this got missed - we're still creating an empty inputarg and boundInputs vector here and transforming it below.